### PR TITLE
External ping method

### PR DESCRIPTION
**Pull request by andreeacalinvtex**

### CODE REVIEW

File: node/clients/pingClient.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  }
+
+  public async ping() {
+    return this.http.get(`${this.baseUrl}/ping`, {cacheable: CacheType.None}).catch(() => Promise.resolve())
+  }
+
+  public static getMiddleware = () => async function ping(ctx: Context, next: () => Promise<any>) {
+    ctx.vtex.logger.info({
+      middleware: 'PING',
+      message: `Pinged`,
```
Comment (tibiiordachevtex) [inline node/clients/pingClient.ts:18]: message: 'Pinged',

Use single quotes '' for strings.
Template literals should contain interpolation or newlines.

File: node/utils/formatError.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+       `
+    }
+  
+    if (error.request) {
+      /*
+       * The request was made but no response was received, `error.request`
+       * is an instance of XMLHttpRequest in the browser and an instance
+       * of http.ClientRequest in Node.js
+       */
+      return `Failed with no response for request: with no response`
```
Comment (tibiiordachevtex) [inline node/utils/formatError.ts:20]: message: 'Failed with no response for request: with no response',

Use single quotes '' for strings.
Template literals should contain interpolation or newlines.

File: node/middlewares/common/isAdminAuthenticated.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { CRON } from '../../utils/constants'
+
+export async function isAdminAuthenticated(
+  ctx: Context,
+  next: () => Promise<any>
+) {
+    interface CtxObject {token: string }
+    try {
+
+      const {token} : CtxObject= await json(ctx.req)
```
Comment (tibiiordachevtex) [inline node/middlewares/common/isAdminAuthenticated.ts:11]: As Airbnb JavaScript Style Guide suggests:

const { token }: CtxObject = await json(ctx.req)

**Review by tibiiordachevtex — COMMENTED** [review]
Just some stylistic suggestions.

File: node/middlewares/scheduler/createScheduler.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        },
+      })
+  
+      ctx.status = 201
+      ctx.body = 'Cron created'
+  
+      await next()
+    } catch (e) {
+      logger.error({
+        middleware: 'CREATE SELLER ACCOUNT NAME CRON',
```
Comment (robertdiaconeasa) [inline node/middlewares/scheduler/createScheduler.ts:30]: Can you please change the string of middleware from 'CREATE SELLER ACCOUNT NAME CRON', into something related with this project, like 'CREATE RETURN APP SCHEDULER CRON'.

File: node/middlewares/scheduler/deleteScheduler.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    try {
+      await scheduler.deleteScheduler(CRON.id)
+  
+      ctx.status = 200
+      ctx.body = 'Cron deleted'
+  
+      await next()
+    } catch (e) {
+      logger.error({
+        middleware: 'DELETE SELLER CRON',
```
Comment (robertdiaconeasa) [inline node/middlewares/scheduler/deleteScheduler.ts:19]: Can you please change the string of middleware from 'DELETE SELLER CRON', into something related with this project, like 'DELETE RETURN APP CRON'.

File: node/utils/constants.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export const CRON = {
+    id: '2943b02a-fd27-47e3-9fcc-9eb00e07e2a0',
+    authToken: '52843179-97f4-4e49-b716-1ecf58809838',
```
Comment (alex-neacsu) [inline node/utils/constants.ts:3]: this is kind of useless if it's added on a public github project 🤷‍♂️ 

A better approach would be to read it from settings

File: node/middlewares/common/isAdminAuthenticated.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+export async function isAdminAuthenticated(
+  ctx: Context,
+  next: () => Promise<any>
+) {
+    interface CtxObject {token: string }
+    try {
+
+      const {token}: CtxObject = await json(ctx.req)
+
+    if (token !== CRON.authToken) {
```
Comment (alex-neacsu) [inline node/middlewares/common/isAdminAuthenticated.ts:13]: would be better to read from settings than a constant

**Review by alex-neacsu — CHANGES_REQUESTED** [review]

File: node/middlewares/common/isAdminAuthenticated.ts (hunk 7 above)
Comment (andreeacalinvtex) [reply node/middlewares/common/isAdminAuthenticated.ts]: done

File: node/utils/constants.ts (hunk 6 above)
Comment (andreeacalinvtex) [reply node/utils/constants.ts]: done

**Review by duxy — APPROVED** [review]

**Review by tibiiordachevtex — APPROVED** [review]

**Pull request closed without merging**